apps/cbportal/base/cloudbroker__cloudspace/methodclass/cloudbroker_cloudspace.py

```diff
@@ -62,40 +65,53 @@ def _destroy(self, cloudspace, reason, ctx):
             machines = self.models.vmachine.search(machine_query)[1:]
             for idx, machine in enumerate(sorted(machines, key=lambda m: m['cloneReference'], reverse=True)):
                 ctx.events.sendMessage(title, 'Deleting Virtual Machine %s/%s' % (idx + 1, len(machines)))
-                j.apps.cloudbroker.machine.destroy(machine['id'], reason)
-                self.cb.machine.destroy_machine(machine['id'], provider)
+                j.apps.cloudbroker.machine.destroy(machine['id'], reason, permanently)
         except:
             cloudspace = self.models.cloudspace.get(cloudspace['id']).dump()
             cloudspace['status'] = status
             self.models.cloudspace.set(cloudspace)
             raise
 
-        # delete routeros
-        ctx.events.sendMessage(title, 'Deleting Virtual Firewall')
-        self._destroyVFW(cloudspace['gid'], cloudspace['id'])
-        cloudspace = self.models.cloudspace.get(cloudspace['id'])
-        cloudspace.status = resourcestatus.Cloudspace.DESTROYED
-        self.cb.cloudspace.release_resources(cloudspace)
+        if not permanently:
+            fwid = '%s_%s' % (cloudspace['gid'], cloudspace['networkId'])
```

Comment:
would be nice to send event here indication you ar estopping VFW like on line 84